### DISCUSSION TITLE
Parameter NUM_PHYS_FUNC and NUM_INTF

### DISCUSSION THREAD

**Discussion by adki** [Xilinx/open-nic-shell#29] [Q&A]
Hi,

Can anyone give comment on these two parameters, 'NUM_PHYS_FUNC' and 'NUM_INTF'?
'NUM_PHYS_FUNC' determines the number of physical function in QDMA system, while 'NUM_INTF' seems the number channels between QDMA system and p2p_250mhz.

Are there any relationship or differences?

What I am trying to figure out is that how to add additional channels through QDMA to carry out additional data processing on the user specific data that is nothing to do with Ethernet packet.

I wonder if there is any material regarding internal structure of OpenNIC, it would be grateful.

Thanks.

**Comment by cneely-amd** [comment]
We have a technical reference manual here: https://github.com/Xilinx/open-nic/blob/main/OpenNIC_manual.pdf

On page 3, it briefly describes NUM_PHYS_FUNC:
![image](https://user-images.githubusercontent.com/84047881/201238006-a85a2945-d097-43ec-a681-bfb58ed38121.png)

I think in `plugin/p2p/box_250mhz/user_plugin_250mhz_inst.vh` NUM_INTF gets set to NUM_PHYS_FUNC on line 35.

**Reply by adki** [reply]
Hi Chris,

I got it and 'NUM_INTF' is the same as 'NUM_PHYS_FUNC'.
By the way, the manual was very helpful to understand Open-NIC, but someone like me could not get details since it just describes an abstract level.
Thanks again.
